lib/trento.pm

```diff
@@ -826,6 +828,43 @@ sub cluster_wait_status {
     die "Timeout waiting for the change" if !$done;
 }
 
+=head3 cluster_wait_status_by_regex
+
+Remotly run 'SAPHanaSR-showAttr' in a loop on $host, wait output that match using regular expression
+=cut
+
+sub cluster_wait_status_by_regex {
+    my ($host, $timeout, $regular_expression, $text_to_find) = @_;
+    $timeout //= bmwqemu::scale_timeout(300);
+    my $_monitor_start_time = time();
+    my $done;
+    while ((time() - $_monitor_start_time <= $timeout) && (!$done)) {
+        sleep 30;
+        my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
```

Comment:
This line can be moved before the `while()`. I don't expect the setting to change inside the loop.

Reply:
ok